pkg/reconciler/v1alpha1/broker/broker.go

```diff
@@ -164,27 +166,30 @@ func (r *reconciler) Reconcile(request reconcile.Request) (reconcile.Result, err
 func (r *reconciler) reconcile(ctx context.Context, b *v1alpha1.Broker) (reconcile.Result, error) {
 	b.Status.InitializeConditions()
 
-	// 1. Channel is created for all events.
+	// 1. Trigger Channel is created for all events. Triggers will Subscribe to this Channel.
 	// 2. Filter Deployment.
 	// 3. Ingress Deployment.
 	// 4. K8s Services that point at the Deployments.
+	// 5. Ingress Channel is created to get events from Triggers back into this Broker via the
```

Comment:
Could we add a note here saying something that this is necessary because subscription can only route to Channel objects and hence we need to create an artificial channel instead of feeding it to the Ingress Service directly. If I understood correctly.

Reply:
Also, probably open an issue to track about relaxing the constraint that only channels can be used as a target for response from a Subscription.

Reply:
Added a note and #925.

Reply:
I think the channel here isn't actually so bad because it decouples the publish and consume failure domains. With the ingress channel, consume doesn't depend on the broker ingress availability, only channel availability (which the consumer already depends on via its subscription to the main trigger channel).

For the intentionally simple "respond with event" use case, having simpler failure modes is good IMO.